bundles/org.openhab.binding.openthermgateway/src/main/java/org/openhab/binding/openthermgateway/handler/OpenThermGatewayHandler.java

```diff
@@ -0,0 +1,273 @@
+
+            if (checkConnection()) {
+                @Nullable
+                OpenThermGatewayConnector conn = connector;
+
+                if (conn != null) {
+                    conn.sendCommand(gatewayCommand);
+                }
+            }
+        }
+    }
+
+    @Override
+    public void connecting() {
+        connecting = true;
+        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.NONE, "Connecting");
+    }
+
+    @Override
+    public void connected() {
+        connecting = false;
+        updateStatus(ThingStatus.ONLINE);
+    }
+
+    @Override
+    public void disconnected() {
+        @Nullable
+        OpenThermGatewayConnector conn = connector;
+
+        @Nullable
+        OpenThermGatewayConfiguration conf = config;
+
+        connecting = false;
+
+        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.NONE, "Disconnected");
+
+        // retry connection if disconnect is not explicitly requested
+        if (conf != null && !explicitDisconnect && conf.connectionRetryInterval > 0) {
+            scheduler.schedule(() -> {
+                if (conn != null && !connecting && !conn.isConnected()) {
+                    connect();
+                }
+            }, conf.connectionRetryInterval, TimeUnit.SECONDS);
+        }
+    }
+
+    @Override
+    public void receiveMessage(Message message) {
+        if (DataItemGroup.dataItemGroups.containsKey(message.getID())) {
+            DataItem[] dataItems = DataItemGroup.dataItemGroups.get(message.getID());
+
+            for (DataItem dataItem : dataItems) {
+                String channelId = dataItem.getSubject();
+
+                if (!OpenThermGatewayBindingConstants.SUPPORTED_CHANNEL_IDS.contains(channelId)) {
+                    continue;
+                }
+
+                State state = null;
+
+                switch (dataItem.getDataType()) {
+                    case FLAGS:
+                        state = OnOffType.from(message.getBit(dataItem.getByteType(), dataItem.getBitPos()));
+                        break;
+                    case UINT8:
+                    case UINT16:
+                        state = new DecimalType(message.getUInt(dataItem.getByteType()));
+                        break;
+                    case INT8:
+                    case INT16:
+                        state = new DecimalType(message.getInt(dataItem.getByteType()));
+                        break;
+                    case FLOAT:
+                        float value = message.getFloat();
+                        @Nullable
+                        Unit<?> unit = dataItem.getUnit();
+                        state = (unit == null) ? new DecimalType(value) : new QuantityType<>(value, unit);
+                        break;
+                    case DOWTOD:
+                        break;
+                }
+
+                if (state != null) {
+                    logger.debug("Received update for channel '{}': {}", channelId, state.toFullString());
```

Comment:
Is the normal String enough? If so please just pass the `state`. Otherways you will also call the `toFullString` when debug-levels do not apply.

Reply:
Removed toFullString